packages/protvista-msa/src/ProtvistaMSA.js

```diff
@@ -1,18 +1,51 @@
 import ProtvistaZoomable from "protvista-zoomable";
 import ReactMSAViewer from "react-msa-viewer";
-import React from "react";
+import React, { useRef, useEffect } from "react";
 import ReactDOM from "react-dom";
 import { select } from "d3";
 
 class ProtvistaMSA extends ProtvistaZoomable {
+  static get properties() {
+    return {
+      onActiveTrackChange: { type: Function },
+    };
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (this.hasAttribute("activeLabel")) {
+      this.activeLabel = this.getAttribute("activeLabel");
+    }
+  }
+
   static get observedAttributes() {
     return ProtvistaZoomable.observedAttributes.concat([
       "labelwidth",
+      "activeLabel",
       "colorscheme",
       "calculateConservation",
     ]);
   }
 
+  get activeLabel() {
+    return this.getAttribute("activeLabel");
+  }
+
+  set activeLabel(value) {
+    this.setAttribute("activeLabel", value);
+  }
+
+  setActiveLabel(newValue) {
```

Comment:
Could the logic in this method go into `set activeLabel`?

Reply:
This could have worked, but since I have moved the label component outside of the main class, it just won't easily work -- `this` and context binding related issues.

But besides that, the `setter` on `activeLabel` is really there just to set the value of a class property, merely a language feature rather a semantic interpretation of what-we-doing-here and in my opinion moving the `setActiveLabel` code to `set activeLabel` is against SRP and possibly now the right place to be also anyway.

I personally wouldn't think an actual logical step like calling a callback function and forcing a refresh would happen inside a setter method, simply because these two are related.

I agree, if this was something we were doing on the value of the label, then it would make sense to put it in its setter method, because we are rather doing something *with* the value of the label, if that makes any sense.

Reply:
As per our discussion, this is not renamed for better semantics.